mathesar/api/viewsets/db_types.py

```diff
@@ -0,0 +1,26 @@
+from django.core.exceptions import ObjectDoesNotExist
+
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from db.types.base import get_available_known_db_types
+
+from mathesar.api.serializers.db_types import DBTypeSerializer
+
+from mathesar.models import Database
+
+
+class DBTypeViewSet(viewsets.ViewSet):
+    def list(self, request):
+        try:
+            db_name = request.query_params['db_name']
```

Comment:
We should be using IDs to refer to API resources, not names. Also, this shouldn't be a required parameter, we should be able to loop through all installed DBs if this is not provided. Filtering by DB ID should be handled similarly to other APIs, see `mathesar/api/filters.py`.

If there's some reason to _require_ a database, then it's better to put this under `/api/v0/databases/<id>/types/`. We should not have multiple patterns to provide a database identifier in the URL.

All of this feedback also applies to the functions API.